### PR TITLE
util: introduce parse_ifindex_or_ifname() and use it in networkctl and resolvectl

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/basic/parse-util.c [hunk 1]
```diff
@@ -93,6 +94,24 @@ int parse_ifindex(const char *s, int *ret) {
         return 0;
 }
 
+int parse_ifindex_or_ifname(const char *s, int *ret) {
+        int r;
+
+        assert(s);
+
```
Comment (ssahani) [inline src/basic/parse-util.c:105]: ``` assert(ret)``` ?

File: src/basic/parse-util.c [hunk 2]
```diff
@@ -93,6 +94,24 @@ int parse_ifindex(const char *s, int *ret) {
...
+        r = parse_ifindex(s, ret);
+        if (r >= 0)
+                return r;
+
```
Comment (ssahani) [inline src/basic/parse-util.c:109]: extra line

**Comment by yuwata** [comment]
@ssahani Thank you for your review. Updated. PTAL.

**Comment by ssahani** [comment]
LGTM

**Comment by yuwata** [comment]
@ssahani Thanks.

**Pull request merged**